app/models/weam.rb

```diff
@@ -163,16 +162,34 @@ def vet_tec_flags_for_approved?
   # If school is a vet tec provider, applicable law code is VET TEC ONLY,
   # and be a non college degree indicator
   def approved?
-    return false if poo_status.blank? || applicable_law_code.blank?
+    return false if applicable_law_code.blank?
 
-    return false unless poo_status =~ Regexp.new('aprvd', 'i')
+    return false unless poo_status_valid?
 
-    # VET TEC ONLY
-    return vet_tec_flags_for_approved? if applicable_law_code =~ Regexp.new(ALCVT, 'i')
+    return vet_tec_approved? if vet_tec?
+
+    return false if invalid_law_code?
 
-    # Other
-    return false if applicable_law_code =~ Regexp.new("#{ALC1}|#{ALC2}", 'i')
     flags_for_approved?
   end
+
+  private
+
+  def poo_status_valid?
+    poo_status =~ Regexp.new('aprvd', 'i')
+  end
+
+  def invalid_law_code?
+    LAW_CODES_BLOCKING_APPROVED_STATUS.none? { |law_code| applicable_law_code.downcase.include? law_code }
```

Comment:
It's possible I'm reading this wrong, but I think this should be `any?`, the opposite of `none?`:
```suggestion
    LAW_CODES_BLOCKING_APPROVED_STATUS.any? { |law_code| applicable_law_code.downcase.include? law_code }
```
And since specs are passing you should probably revisit the `approved?` specs to make sure you're covering all cases.

In the spec file, lines [219 and 220](https://github.com/department-of-veterans-affairs/gibct-data-service/blob/c55f1d63a8b769681404fcf244063dee306b9289/spec/models/weam_spec.rb#L220) are building instances with the exact same attributes.  You can remove one of those.

`invalid_law_code?` evaluates to true with this definition.
Your spec is still passing because `flags_for_approved?` evaluates to `false`.  
Make sure you have a spec where `flags_for_approved?` will evaluate to `true` to be sure that `invalid_law_code?` is working.
e.g. change line 219 from `let(:non_approved_law_code) { build :weam, :approved_poo_and_non_approved_law_code }` to `let(:non_approved_law_code) { build :weam, :approved_poo_and_non_approved_law_code, ojt_indicator: true }` and see what happens.

Reply:
Good catch, thanks!